File: clone.py
#!/bin/python

import requests
from pprint import pprint
from dotenv import dotenv_values

config = dotenv_values('.env')

def doRequest(payload):
    response = requests.get('https://api.github.com/user/repos', params=payload, auth=(config['USERNAME'],config['TOKEN']))
    for repo in response.json():
        print(repo['language'] if repo['language'] else 'Others', end='\t') 
        print(repo['ssh_url'])
    return len(response.json())

items = doRequest({'page': '0', 'per_page': '100'})

while items > 100:
    items = doRequest({'page': items, 'per_page': '100'})

File: clone.php
<?php

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, "https://api.github.com/user/repos?per_page=200");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_USERAGENT, "Mozilla/5.0 (X11; Linux x86_64; rv:131.0) Gecko/20100101 Firefox/131.0");
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Authorization: Bearer '. $_SERVER['TOKEN'], 
    'X-GitHub-Api-Version: 2022-11-28'
]);

$json = curl_exec($ch);

$data = json_decode($json, true);

foreach($data as $repo) {
    echo (empty($repo['language']) ? 'Unknown' : $repo['language']) . ' ' . $repo['ssh_url'] . PHP_EOL;
}
